### PR TITLE
feat: add fashion query language example 

### PR DESCRIPTION
**Pull request by theUnkownName**

### CODE REVIEW

File: fashion-example-query/fashion/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Star Trek QueryLang
```
Comment (JoanFM) [inline fashion-example-query/fashion/README.md:1]: do not forget before merge to change this README

File: fashion-example-query/fashion/app.py [hunk 2]
```diff
@@ -67,14 +79,23 @@ def download_data(targets, download_proxy=None):
...
+            else:
+                v['data'] = load_mnist(v['filename'])
 
 
 def index():
+    for j in range(num_docs):
+        d = jina_pb2.Document()
+        d.embedding.CopyFrom(array2pb(targets['index']['data'][j]))
+        label = (targets['labels']['data'][j]).item()
+        d.tags.update({'id': label})
```
Comment (JoanFM) [inline fashion-example-query/fashion/app.py:93]: d.tags.update({'label': label}). We do not want to use `id` but label. And previously map `label_id` to `label name` as in zalandoresearch repo

File: fashion-example-query/fashion/app.py [hunk 3]
```diff
@@ -110,6 +129,10 @@ def config():
         'query': {
             'url': 'http://fashion-mnist.s3-website.eu-central-1.amazonaws.com/t10k-images-idx3-ubyte.gz',
             'filename': os.path.join('./workspace', 'query-original')
+        },
+        'labels': {
```
Comment (JoanFM) [inline fashion-example-query/fashion/app.py:133]: consider that you will need `index-labels` and `query-labels`

File: fashion-example-query/fashion/flows/index.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+!Flow
```
Comment (JoanFM) [inline fashion-example-query/fashion/flows/index.yml:1]: this file is to be removed

File: fashion-example-query/fashion/flows/query.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+!Flow
```
Comment (JoanFM) [inline fashion-example-query/fashion/flows/query.yml:1]: to be removed

File: fashion-example-query/fashion/pods/encode.yml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (JoanFM) [inline fashion-example-query/fashion/pods/encode.yml:1]: to be removed

File: fashion-example-query/fashion/pods/doc.yml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+!BinaryPbIndexer
```
Comment (JoanFM) [inline fashion-example-query/fashion/pods/doc.yml:1]: to be removed

File: fashion-example-query/fashion/pods/craft.yml [hunk 8]
```diff
@@ -0,0 +1,9 @@
+!Sentencizer
```
Comment (JoanFM) [inline fashion-example-query/fashion/pods/craft.yml:1]: to be removed

File: fashion-example-query/fashion/pods/chunk.yml [hunk 9]
```diff
@@ -0,0 +1,17 @@
+!CompoundIndexer
```
Comment (JoanFM) [inline fashion-example-query/fashion/pods/chunk.yml:1]: to be removed

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: fashion-example-query/fashion/app.py [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+            else:
+                v['data'] = load_mnist(v['filename'])
+
+
+def index():
+    for j in range(num_docs):
+        d = jina_pb2.Document()
+        d.embedding.CopyFrom(array2pb(targets['index']['data'][j]))
+        label = (targets['labels']['data'][j]).item()
+        d.tags.update({'tags': label})
```
Comment (JoanFM) [inline fashion-example-query/fashion/app.py]:
```suggestion
        d.tags.update({'label': get_mapped_label[label]})
```

File: fashion-example-query/fashion/app.py [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+                v['data'] = load_labels(v['filename'])
+            else:
+                v['data'] = load_mnist(v['filename'])
+
+
+def index():
+    for j in range(num_docs):
+        d = jina_pb2.Document()
+        d.embedding.CopyFrom(array2pb(targets['index']['data'][j]))
+        label = (targets['labels']['data'][j]).item()
```
Comment (JoanFM) [inline fashion-example-query/fashion/app.py]: Shouldn't it work like this?
```suggestion
        label = targets['labels']['data'][j][0]
```

File: fashion-example-query/fashion/flow-query.yml [hunk 12]
```diff
@@ -0,0 +1,16 @@
+!Flow
+with:
+  read_only: true  # better add this in the query time
+  logserver: $WITH_LOGSERVER
+  compress_hwm: 1024
+pods:
+  encode:
+    uses: encode/encode.yml
+    parallel: $PARALLEL
+  index:
```
Comment (JoanFM) [inline fashion-example-query/fashion/flow-query.yml:10]: u will have to create 10 `parallel` indexers like this. In each of this, you have to put a `FilterQL` to ensure that it receives the documents with the `required` label

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: fashion-example-query/fashion/app.py (hunk 10 above)
Comment (theUnkownName) [reply fashion-example-query/fashion/app.py]: Can I do that? I get an error saying the `function' object is not subscriptable`

Comment (theUnkownName) [reply fashion-example-query/fashion/app.py]: Ah never mind

Comment (JoanFM) [reply fashion-example-query/fashion/app.py]: Sorry wrong brackests, it is a call to function not subsribtion

Comment (JoanFM) [reply fashion-example-query/fashion/app.py]: corrected suggestion

Comment (theUnkownName) [reply fashion-example-query/fashion/app.py]: Yeah took me some time to see it, thanks

File: fashion-example-query/README.md [hunk 13]
```diff
@@ -29,25 +31,71 @@ Now that we've got the code to load our data, we're going to dive into writing o
...
-
+This is an example of the results after the query using all categories
 
 ![alt text](results.png "Results")
 
 
+## QueryLanguage
+
+So the real magic here, if you compare it with the original HelloWorld, is that we are using QueryLanguage.
+This is happening in the index flow of each category, for example, for the ```indexer-dress.yml``` we have
```
Comment (JoanFM) [inline fashion-example-query/README.md:92]: it also happens in the query flow.

File: fashion-example-query/README.md [hunk 14]
```diff
@@ -14,6 +14,8 @@
 First of all, read up on [Jina 101](https://github.com/jina-ai/jina/tree/master/docs/chapters/101) so you have a clear understanding of how Jina works. We're going to refer to those concepts a lot. We assume you already have some knowledge of Python and machine learning.
 
 This is an extension of the Hello World example, but here we use QueryLanguage to filter the results by category.
```
Comment (JoanFM) [inline fashion-example-query/README.md:16]: we use QueryLanguage to create an index per category, thus having results filtered by categories

File: fashion-example-query/fashion/app.py [hunk 15]
```diff
@@ -94,10 +93,11 @@ def load_mnist(path):
 
 def load_labels(path):
     with gzip.open(path, 'rb') as fp:
-        return np.frombuffer(fp.read(), dtype=np.uint8, offset=16).reshape([-1, 1])
+        return np.frombuffer(fp.read(), dtype=np.uint8, offset=8).reshape([-1, 1])
 
 
 def download_data(target, download_proxy=None):
+    labels_len = -1
```
Comment (JoanFM) [inline fashion-example-query/fashion/app.py:100]: remove

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: fashion-example-query/fashion/Dockerfile [hunk 16]
```diff
@@ -0,0 +1,10 @@
+FROM jinaai/jina:0.5.0
+
+COPY . /workspace
+WORKDIR /workspace
+
+RUN apt-get update && pip install -r requirements.txt
+
+RUN python app.py dryrun
```
Comment (JoanFM) [inline fashion-example-query/fashion/Dockerfile:8]: RUN python app.py index

File: fashion-example-query/fashion/Dockerfile [hunk 17]
```diff
@@ -0,0 +1,10 @@
...
+
+ENTRYPOINT ["python", "app.py"]
```
Comment (JoanFM) [inline fashion-example-query/fashion/Dockerfile:10]: ENTRYPOINT ["python", "app.py", "query"]

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Review by JoanFM — APPROVED** [review]

**Pull request merged**